internal/project/project.go

```diff
@@ -462,6 +486,7 @@ func (p *Project) updateGraph() bool {
 
 	start := time.Now()
 	p.Log("Starting updateGraph: Project: " + p.name)
+	p.host.fs.ClearCache()
```

Comment:
I think we should keep the cache only for updateGraph duration as otherwise we dont know if its correct or not.

Reply:
you probably saw this with typings installer tests but in practice this could be watch events as well

Reply:
I'm not sure where you're telling me to put the call; I don't know where this would go otherwise.

Are you saying to clear the cache at the end of updateGraph?

I can do it that way, but I don't think that would affect ATA, since that code can happen any time concurrently (part of the reason why I am not giving it a cached FS, and plan to decouple it from the Project in a followup to fix the races)

Reply:
i meant to keep the cache only for duration of update graph " so uncached before and after that call. (eg of how we did this in strada - this ensures the correctness if for some reason we need to check with FS on file existence etc  https://github.com/microsoft/TypeScript/blob/main/src/services/services.ts#L1782 where cache lives only for program creation and then goes away

Reply:
So, defer this call so it runs when updateGraph returns?

Reply:
Updated, PTAL